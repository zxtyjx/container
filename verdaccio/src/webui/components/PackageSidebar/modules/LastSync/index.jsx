import React from 'react';
import propTypes from 'prop-types';
import Module from '../../Module';
import ModuleContentPlaceholder from '../../ModuleContentPlaceholder';

import classes from './style.scss';

const renderRecentReleases = (recentReleases) => {
  return (
    <ul>
      {recentReleases.map((versionInfo) => {
        const {version, time} = versionInfo;
        return (
          <li className="last-sync-item" key={version}>
            <span>{version}</span>
            <span>{time}</span>
          </li>
        );
      })}
    </ul>
  );
};

const LastSync = ({recentReleases = [], lastUpdated = ''}) => {
  return (
    <Module
      title="Last Sync"
      description={lastUpdated}
      className={classes.releasesModule}
    >
      {recentReleases.length ? (
        renderRecentReleases(recentReleases)
      ) : (
        <ModuleContentPlaceholder text="Not Available!" />
      )}
    </Module>
  );
};

LastSync.propTypes = {
  recentReleases: propTypes.array,
  lastUpdated: propTypes.string
};

export default LastSync;
