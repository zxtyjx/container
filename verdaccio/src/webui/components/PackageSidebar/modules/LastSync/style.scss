.releasesModule {
  li {
    display: flex;
    font-size: 14px;
    line-height: 2;
    span:last-child {
      margin-left: auto;
    }
  }
}
