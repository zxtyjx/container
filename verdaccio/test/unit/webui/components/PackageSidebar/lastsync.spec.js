/**
 * LastSync component
 */

import React from 'react';
import { shallow } from 'enzyme';
import LastSync from '../../../../../src/webui/components/PackageSidebar/modules/LastSync/index';

describe('<PackageSidebar /> : <LastSync />', () => {
  it('should check the default props condition', () => {
    const wrapper = shallow(<LastSync/>);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should load the LastSync component and match snapshot', () => {
    const props = {
      lastUpdated: '2017/12/14, 15:43:52',
      recentReleases: [
        { time: '2017/12/14, 15:43:27', version: '2.7.1' },
        { time: '2017/12/05, 23:25:06', version: '2.7.0' },
        { time: '2017/11/08, 22:47:16', version: '2.6.6' }
      ]
    };
    const wrapper = shallow(<LastSync {...props}/>);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
