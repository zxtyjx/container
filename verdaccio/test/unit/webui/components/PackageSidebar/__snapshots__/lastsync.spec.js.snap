// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PackageSidebar /> : <LastSync /> should check the default props condition 1`] = `"<div class=\\"module releasesModule\\"><h2 class=\\"moduleTitle\\">Last Sync</h2><div><p class=\\"emptyPlaceholder\\">Not Available!</p></div></div>"`;

exports[`<PackageSidebar /> : <LastSync /> should load the LastSync component and match snapshot 1`] = `"<div class=\\"module releasesModule\\"><h2 class=\\"moduleTitle\\">Last Sync<span>2017/12/14, 15:43:52</span></h2><div><ul><li class=\\"last-sync-item\\"><span>2.7.1</span><span>2017/12/14, 15:43:27</span></li><li class=\\"last-sync-item\\"><span>2.7.0</span><span>2017/12/05, 23:25:06</span></li><li class=\\"last-sync-item\\"><span>2.6.6</span><span>2017/11/08, 22:47:16</span></li></ul></div></div>"`;
